datamol/io.py

```diff
@@ -95,21 +97,30 @@ def read_sdf(
             `as_df` is True.
         include_computed: Include computed properties in the columns.  Only relevant if
             `as_df` is True.
+        sanitize: Whether to sanitize the molecules
+        strict_parsing: If set to false, the parser is more lax about correctness of the contents.
     """
 
     # File-like object
     if isinstance(urlpath, io.IOBase):
-        supplier = Chem.ForwardSDMolSupplier(urlpath)
+        supplier = Chem.ForwardSDMolSupplier(urlpath, sanitize=False, strictParsing=strict_parsing)
         mols = [mol for mol in supplier if mol is not None]
 
     # Regular local or remote paths
     else:
         with fsspec.open(urlpath) as f:
             if str(urlpath).endswith(".gz") or str(urlpath).endswith(".gzip"):
                 f = gzip.open(f)
-            supplier = Chem.ForwardSDMolSupplier(f)
+            supplier = Chem.ForwardSDMolSupplier(f, sanitize=False, strictParsing=strict_parsing)
             mols = [mol for mol in supplier if mol is not None]
 
+    if sanitize == True:
+        mols = [
+            dm.set_mol_props(dm.sanitize_mol(mol), mol.GetPropsAsDict())
```

Comment:
`sanitize_mol` Can return None, thus `set_mol_props` can fail.

Reply:
Have a look now, I added a check alongside the mol check